pkg/local_object_storage/metabase/inhume.go

```diff
@@ -156,20 +175,19 @@ func (db *DB) Inhume(prm InhumePrm) (res InhumeRes, err error) {
 				if err != nil {
 					return err
 				}
-			} else {
-				// garbage object can probably lock some objects, so they should become
-				// unlocked after its decay
-				err = freePotentialLocks(tx, cnr, id)
-				if err != nil {
-					return fmt.Errorf("free potential locks: %w", err)
-				}
 			}
 
 			// consider checking if target is already in graveyard?
 			err = bkt.Put(targetKey, value)
 			if err != nil {
 				return err
 			}
+
+			if prm.lockObjectHandling {
+				if isLockObject(tx, cnr, id) {
```

Comment:
`if prm.lockObjectHandling && isLockObject(...) {` ?

Reply:
hm, i have pushed a strange version, do not know how that happened: it even does not contain comments, mb rebasing problems

pushed correct version